hive/lib/src/adapters/date_time_adapter.dart

```diff
@@ -1,18 +1,42 @@
 import 'package:hive/hive.dart';
 
 /// Adapter for DateTime
-class DateTimeAdapter extends TypeAdapter<DateTime> {
+class DateTimeAdapter extends TypeAdapter<_DateTimeWithoutTZ> {
   @override
   final typeId = 16;
 
+  @override
+  _DateTimeWithoutTZ read(BinaryReader reader) {
+    var micros = reader.readInt();
+    return _DateTimeWithoutTZ.fromMillisecondsSinceEpoch(micros);
+  }
+
+  @override
+  void write(BinaryWriter writer, DateTime obj) {
+    writer.writeInt(obj.millisecondsSinceEpoch);
+  }
+}
+
+class _DateTimeWithoutTZ extends DateTime {
+  _DateTimeWithoutTZ.fromMillisecondsSinceEpoch(int millisecondsSinceEpoch)
+      : super.fromMillisecondsSinceEpoch(millisecondsSinceEpoch);
+}
+
+/// Alternative adapter for DateTime with time zone info
+class DateTimeWithTimezoneAdapter extends TypeAdapter<DateTime> {
```

Comment:
I think it is not necessary to store `isUtc`.

I would rather use the following code:
```dart
class DateTimeWithTimezoneAdapter extends TypeAdapter<DateTime> {
  @override
  final typeId = 18;

  @override
  DateTime read(BinaryReader reader) {
    var millis = reader.readInt();
    return DateTime.fromMillisecondsSinceEpoch(millis, isUtc: true).toLocal();
  }

  @override
  void write(BinaryWriter writer, DateTime obj) {
    writer.writeInt(obj.toUtc().millisecondsSinceEpoch);
  }
}
```

Reply:
I store `isUtc` value to make sure returned DateTime from `read` is same kind as the value previously written. I think otherwise some weird things might happen or confuse users if they try to write DateTime in UTC and get in local timezone.

Reply:
Yes you are right! Leave it as it is then :+1: